tests/10-api.spec.js

```diff
@@ -1,18 +1,292 @@
+import {ZcapClient} from '@digitalbazaar/ezcap';
+import {httpClient, DEFAULT_HEADERS} from '@digitalbazaar/http-client';
+import {securityLoader} from '@digitalbazaar/security-document-loader';
+import zcapCtx from 'zcap-context';
+import {authorizeZcapInvocation} from '../lib';
 
+const didKeyDriver = didKey.driver();
+const loader = securityLoader();
+loader.addStatic(zcapCtx.CONTEXT_URL, zcapCtx.CONTEXT);
+
+const documentLoader = loader.build();
+
+const TEST_SERVER_PORT = 5000;
+
+function _startServer({app, port = TEST_SERVER_PORT}) {
+  return new Promise(resolve => {
+    const server = app.listen(port, () => {
+      console.log(`Test server listening at http://localhost:${port}`);
+      return resolve(server);
+    });
+  });
+}
+
+const app = express();
+app.use(express.json());
+
+// mount the test routes
+app.post('/documents',
+  authorizeZcapInvocation({
+    documentLoader,
+    getExpectedTarget() {
+      return {
+        expectedTarget: ['http://localhost:5000/documents']
+      };
+    },
+    getRootController() {
+      // root controller(Admin DID)
+      return 'did:key:z6Mkfeco2NSEPeFV3DkjNSabaCza1EoS3CmqLb1eJ5BriiaR';
+    },
+    expectedHost: 'localhost:5000'
+  }),
+  // eslint-disable-next-line no-unused-vars
+  (req, res, next) => {
+    res.json({message: 'Post request was successful.'});
+  });
+
+app.get('/test/:id',
+  authorizeZcapInvocation({
+    documentLoader,
+    getRootController() {
+      // root controller(Admin DID)
+      return 'did:key:z6Mkfeco2NSEPeFV3DkjNSabaCza1EoS3CmqLb1eJ5BriiaR';
+    },
+    getExpectedTarget({req}) {
+      const expectedTarget =
+        `http://localhost:5000/documents/${encodeURIComponent(req.params.id)}`;
+      // intentionally set return value to not be an object
+      return expectedTarget;
+    },
+    expectedHost: 'localhost:5000'
+  }),
+  // eslint-disable-next-line no-unused-vars
+  (req, res, next) => {
+    res.json({message: 'Get request was successful.'});
+  });
+
+// eslint-disable-next-line no-unused-vars
+app.use(function(err, req, res, next) {
+  res.status(500).send({message: err.message, name: err.name});
+});
+
+let server;
+before(async () => {
+  server = await _startServer({app});
+  // do other stuff that you need to, or simply
+  // return _startServer({app});
+});
+
+after(async () => {
+  server.close();
+});
 describe('ezcap-express', () => {
-  describe('verify proper exports', () => {
-    it(`should properly export authorize`, async () => {
-      should.exist(authorizeZcapInvocation);
-      authorizeZcapInvocation.should.be.a('function');
-    });
-    it(`should properly export functions from main.js`, async () => {
-      should.exist(middleware.authorizeZcapInvocation);
-      middleware.authorizeZcapInvocation.should.be.a('function');
+  describe('authorizeZcapInvocation', () => {
+    it('should succeed if correct data is passed', async () => {
+      const url = 'http://localhost:5000/documents';
```

Comment:
We should use a constant at the top for the base URL here and do something like this:

```suggestion
      const url = `${BASE_URL}/documents`;
```

Reply:
Updated b078468